src/Ng/Jobs/Catalog2PackageFixupJob.cs

```diff
@@ -0,0 +1,183 @@
+        private static readonly TimeSpan MaximumPackageProcessingTime = TimeSpan.FromMinutes(10);
+        private static readonly TimeSpan DonePollingInterval = TimeSpan.FromMinutes(1);
+
+        private IServiceProvider _serviceProvider;
+
+        public Catalog2PackageFixupJob(ITelemetryService telemetryService, ILoggerFactory loggerFactory)
+            : base(telemetryService, loggerFactory)
+        {
+            ThreadPool.SetMinThreads(MaxDegreeOfParallelism, 4);
+        }
+
+        public override string GetUsage()
+        {
+            return "Usage: ng catalog2packagefixup"
+                   + $"-{Arguments.Source} <catalog>"
+                   + $"-{Arguments.Verify} true/false"
+                   + $"-{Arguments.StorageAccountName} <azure-account>"
+                   + $"-{Arguments.StorageKeyValue} <azure-key> "
+                   + $"-{Arguments.StorageContainer} <azure-container>";
+        }
+
+        protected override void Init(IDictionary<string, string> arguments, CancellationToken cancellationToken)
+        {
+            var source = arguments.GetOrThrow<string>(Arguments.Source);
+            var storageAccount = arguments.GetOrThrow<string>(Arguments.StorageAccountName);
+            var storageKey = arguments.GetOrThrow<string>(Arguments.StorageKeyValue);
+            var storageContainer = arguments.GetOrThrow<string>(Arguments.StorageContainer);
+            var verify = arguments.GetOrDefault(Arguments.Verify, false);
+            var verbose = arguments.GetOrDefault(Arguments.Verbose, false);
+
+            var services = new ServiceCollection();
+
+            services.AddSingleton(TelemetryService);
+            services.AddSingleton(LoggerFactory);
+            services.AddLogging();
+
+            // Prepare the HTTP Client
+            services.AddSingleton(p =>
+            {
+                var assembly = Assembly.GetEntryAssembly();
+                var assemblyName = assembly.GetName().Name;
+                var assemblyVersion = assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion ?? "0.0.0";
+
+                var httpClient = new HttpClient(new WebRequestHandler
+                {
+                    AllowPipelining = true
+                });
+
+                httpClient.DefaultRequestHeaders.Add("User-Agent", $"{assemblyName}/{assemblyVersion}");
+                httpClient.Timeout = TimeSpan.FromMinutes(10);
+
+                return httpClient;
+            });
+
+            // Prepare the catalog reader.
+            services.AddSingleton(p =>
+            {
+                var telemetryService = p.GetRequiredService<ITelemetryService>();
+                var httpMessageHandlerFactory = CommandHelpers.GetHttpMessageHandlerFactory(telemetryService, verbose);
+
+                return new CollectorHttpClient(httpMessageHandlerFactory());
+            });
+
+            services.AddTransient(p =>
```

Comment:
In Autofac you could simplify this by explicitly specifying the `new Uri(source)` parameter...can you do this with `ServiceCollection`?

Reply:
Ah, this is a good point. Usually in ASP.NET Core world you have a type that represents your config/options that you inject into your service. So my options are:

1. Bring in Autofac to configure this service. That seems a little overkill
2. I create a new option type to configure the `CatalogIndexReader`. This also seems a little overkill

Do you feel strongly about this? If so, I'll go with option 2. Otherwise I'll leave it as is :)

Reply:
Oh it was definitely a nitpick. This is fine. Was just curious about ASP.NET Core.